xbmc/platform/darwin/ios/IOSEAGLView.mm

```diff
@@ -88,44 +88,13 @@ - (void)layoutSubviews
   }
 }
 
-- (CGFloat) getScreenScale:(UIScreen *)screen
-{
-  CGFloat ret = 1.0;
-  if ([screen respondsToSelector:@selector(scale)])
-  {
-    // normal other iDevices report 1.0 here
-    // retina devices report 2.0 here
-    // this info is true as of 19.3.2012.
-    if([screen scale] > 1.0)
-    {
-      ret = [screen scale];
-    }
-
-    //if no retina display scale detected yet -
-    //ensure retina resolution on supported devices mainScreen
-    //even on older iOS SDKs
-    double screenScale = 1.0;
-    if (ret == 1.0 && screen == [UIScreen mainScreen] && CDarwinUtils::DeviceHasRetina(screenScale))
-    {
-      ret = screenScale;//set scale factor from our static list in case older SDKs report 1.0
-    }
-
-    // fix for ip6 plus which seems to report 2.0 when not compiled with ios8 sdk
-    if (CDarwinUtils::DeviceHasRetina(screenScale) && screenScale == 3.0)
-    {
-      ret = screenScale;
-    }
-  }
-  return ret;
-}
-
 - (void) setScreen:(UIScreen *)screen withFrameBufferResize:(BOOL)resize
 {
   CGFloat scaleFactor = 1.0;
   CAEAGLLayer *eaglLayer = (CAEAGLLayer *)[self layer];
 
   currentScreen = screen;
-  scaleFactor = [self getScreenScale: currentScreen];
+  scaleFactor = [currentScreen scale];
```

Comment:
You can’t do that ... the hardcoded List is needed because the scale method will give wrong results in the case that we compile with an not up to date iOS SDK and run on new devices that were released after the used SDK.
In case of the newer iPads when compiling with Xcode 9 for example - scale will return 1 instead of 2 or 3.
So we use that hardcoded type list and scales to support full retina solution on newer devices.
Alternative would be to always bump to newest Xcode and sdk whenever apple releases new devices.
I can‘t approve removing this. ;)

Reply:
I see now. Thank you for the explanation! Will revert.

Reply:
Once that one commit is dropped i am in general fine with this.

Reply:
Maybe using (nativeScale)[https://developer.apple.com/documentation/uikit/uiscreen/1617825-nativescale] would work? Can't test atm, will check later.

Reply:
You can believe me that I tested the obvious solutions back then. The behavior I described was even documented by apple. Those Methode return compatibility values when devices are to new for the used SDK.